custom_components/smartthinq_sensors/wideq/range.py

```diff
@@ -0,0 +1,207 @@
+"""------------------for Oven"""
+import enum
+import logging
+
+from typing import Optional
+
+from . import (
+    FEAT_COOKTOP_STATE,
+    FEAT_COOKTOP_LEFT_FRONT_STATE,
+    FEAT_COOKTOP_LEFT_REAR_STATE,
+    FEAT_COOKTOP_CENTER_STATE,
+    FEAT_COOKTOP_RIGHT_FRONT_STATE,
+    FEAT_COOKTOP_RIGHT_REAR_STATE,
+    FEAT_OVEN_STATE,
+    FEAT_OVEN_LOWER_STATE,
+    FEAT_OVEN_UPPER_STATE,
+)
+
+from .device import (
+    Device,
+    DeviceStatus,
+    UNITTEMPMODES,
+    UNIT_TEMP_FAHRENHEIT,
+    UNIT_TEMP_CELSIUS,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+OVEN_TEMP_UNIT = {
+    "0": UNITTEMPMODES.Fahrenheit,
+    "1": UNITTEMPMODES.Celsius,
+}
+
+
+class RangeDevice(Device):
+    """A higher-level interface for a cooking range."""
+
+    def __init__(self, client, device):
+        super().__init__(client, device, RangeStatus(self, None))
+
+    def reset_status(self):
+        self._status = RangeStatus(self, None)
+        return self._status
+
+    def poll(self) -> Optional["RangeStatus"]:
+        """Poll the device's current state."""
+
+        res = self.device_poll("ovenState")
+        if not res:
+            return None
+
+        self._status = RangeStatus(self, res)
+        return self._status
+
+
+class RangeStatus(DeviceStatus):
+    """Higher-level information about an range's current status.
+
+    :param device: The Device instance.
+    :param data: JSON data from the API.
+    """
+
+    def __init__(self, device, data):
+        super().__init__(device, data)
+        self._oven_temp_unit = None
+
+    def _update_features(self):
+        result = [
+            self.cooktop_left_front_state,
+            self.cooktop_left_rear_state,
+            self.cooktop_center_state,
+            self.cooktop_right_front_state,
+            self.cooktop_right_rear_state,
+            self.oven_lower_state,
+            self.oven_lower_target_temp,
+            self.oven_upper_state,
+            self.oven_upper_target_temp,
+            self.oven_temp_unit,
+        ]
+        return
+
+    def _get_oven_temp_unit(self):
+        if not self._oven_temp_unit:
+            oven_temp_unit = self.lookup_enum(["MonTempUnit"])
+            if not oven_temp_unit:
+                self._oven_temp_unit = STATE_OPTIONITEM_NONE
+            else:
+                self._oven_temp_unit = (
+                    OVEN_TEMP_UNIT.get(oven_temp_unit, UNITTEMPMODES.Celsius)
+                ).value
+        return self._oven_temp_unit
+
+    @property
+    def is_on(self):
+        return self.is_cooktop_on or self.is_oven_on
+
+    @property
+    def is_cooktop_on(self):
+        result = [
+            self.cooktop_left_front_state,
+            self.cooktop_left_rear_state,
+            self.cooktop_center_state,
+            self.cooktop_right_front_state,
+            self.cooktop_right_rear_state,
+        ]
+        for r in result:
+            if r != 'Off':
```

Comment:
Now this can be changed in this way:

```
if r and r != STATE_OPTIONITEM_OFF:
```

so that this not depend on the translation for the enum.

Reply:
Done.